src/naming/main.cairo

```diff
@@ -527,6 +531,21 @@ func set_l1_contract{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check
     return ();
 }
 
+@external
+func set_referral_contract{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
+    address: felt
+) {
+    // Verify that caller is admin
+    let (caller) = get_caller_address();
+    let (admin_address) = _admin_address.read();
+    assert caller = admin_address;
```

Comment:
Don't we have a function to check admin ? it seems that we'll reuse several times if we do

Reply:
Not at the moment, should I add one ?

Reply:
I think yes if the code is used in another function like this we can delete some code